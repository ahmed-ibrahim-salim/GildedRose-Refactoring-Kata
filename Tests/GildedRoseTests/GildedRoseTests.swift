import ApprovalTests_Swift
@testable import GildedRose
import XCTest

class GildedRoseTests: XCTestCase {
    func testFoo() throws {
        // try Approvals.verify(app.items[0].name)
        let names = ["new name", "Aged Brie",
                     "Backstage passes to a TAFKAL80ETC concert",
                     "Sulfuras, Hand of Ragnaros"]
        let sellIns = [-1, 0, 2, 6, 11]
        let qualities = [0, 1, 49, 50]

        try CombinationApprovals.verifyAllCombinations(makeGildedRose, names, sellIns, qualities)
    }
}

// MARK: Helpers

extension GildedRoseTests {
    func makeGildedRose(name: String, sellIn: Int, quality: Int) -> GildedRose {
        let items = [Item(name: name, sellIn: sellIn, quality: quality)]
        let app = GildedRose(items: items)
        app.updateQuality()
        return app
    }
}
