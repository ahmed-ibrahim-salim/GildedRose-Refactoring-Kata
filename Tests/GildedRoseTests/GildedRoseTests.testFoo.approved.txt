["Aged Brie", -1, 0] -> GildedRose.GildedRose
["Aged Brie", -1, 1] -> GildedRose.GildedRose
["Aged Brie", -1, 49] -> GildedRose.GildedRose
["Aged Brie", -1, 50] -> GildedRose.GildedRose
["Aged Brie", 0, 0] -> GildedRose.GildedRose
["Aged Brie", 0, 1] -> GildedRose.GildedRose
["Aged Brie", 0, 49] -> GildedRose.GildedRose
["Aged Brie", 0, 50] -> GildedRose.GildedRose
["Aged Brie", 2, 0] -> GildedRose.GildedRose
["Aged Brie", 2, 1] -> GildedRose.GildedRose
["Aged Brie", 2, 49] -> GildedRose.GildedRose
["Aged Brie", 2, 50] -> GildedRose.GildedRose
["Aged Brie", 6, 0] -> GildedRose.GildedRose
["Aged Brie", 6, 1] -> GildedRose.GildedRose
["Aged Brie", 6, 49] -> GildedRose.GildedRose
["Aged Brie", 6, 50] -> GildedRose.GildedRose
["Aged Brie", 11, 0] -> GildedRose.GildedRose
["Aged Brie", 11, 1] -> GildedRose.GildedRose
["Aged Brie", 11, 49] -> GildedRose.GildedRose
["Aged Brie", 11, 50] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", -1, 0] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", -1, 1] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", -1, 49] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", -1, 50] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 0, 0] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 0, 1] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 0, 49] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 0, 50] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 2, 0] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 2, 1] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 2, 49] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 2, 50] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 6, 0] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 6, 1] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 6, 49] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 6, 50] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 11, 0] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 11, 1] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 11, 49] -> GildedRose.GildedRose
["Backstage passes to a TAFKAL80ETC concert", 11, 50] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", -1, 0] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", -1, 1] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", -1, 49] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", -1, 50] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 0, 0] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 0, 1] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 0, 49] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 0, 50] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 2, 0] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 2, 1] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 2, 49] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 2, 50] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 6, 0] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 6, 1] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 6, 49] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 6, 50] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 11, 0] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 11, 1] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 11, 49] -> GildedRose.GildedRose
["Sulfuras, Hand of Ragnaros", 11, 50] -> GildedRose.GildedRose
